pandas/core/reshape/reshape.py

```diff
@@ -395,15 +395,29 @@ def pivot(self, index=None, columns=None, values=None):
     See DataFrame.pivot
     """
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        if index is None:
+            cols = [columns]
```

Comment:
do we have tests for each one of these paths here? e.g. branching a lot more now

Reply:
Yes, this is an enhancement. I have added the test cases and need to update the docs. Thanks for your time.